### PR TITLE
nemu-config: adjust defaults in configuration-nemu.toml

### PR DESCRIPTION
**Pull request by egernst**
- For initial offering of virtiofsd, hugepages are required
- use the qemu hypervisor type for configuration template
- decrease virtiofs cache size from 8192 to 1024
- For initial experimental release, utilize virtio-serial instead of
vsock

Fixes: # 1662

Signed-off-by: Eric Ernst <eric.ernst@intel.com>

### CODE REVIEW

File: cli/config/configuration-nemu.toml.in [hunk 1]
```diff
@@ -187,7 +187,7 @@ enable_iothreads = @DEFENABLEIOTHREADS@
 # If true and vsocks are supported, use vsocks to communicate directly
 # with the agent (no proxy is started).
 # Default true
-use_vsock = true
+#use_vsock = true
```
Comment (ganeshmaharaj) [inline cli/config/configuration-nemu.toml.in]: Does this have to be commented out?

Comment (egernst) [reply cli/config/configuration-nemu.toml.in]: this matches how it is handled in QEMU

Comment (ganeshmaharaj) [reply cli/config/configuration-nemu.toml.in]: aah. well given that default is true in itself it doesn't matter.

File: Makefile [hunk 2]
```diff
@@ -163,9 +163,9 @@ DEFENTROPYSOURCE := /dev/urandom
 DEFDISABLEBLOCK := false
 DEFSHAREDFS := virtio-9p
 DEFSHAREDFS_NEMU := virtio-fs
-DEFVIRTIOFSDAEMON := $(VIRTIOFSDBINDIR)/virtiofsd
+DEFVIRTIOFSDAEMON := $(VIRTIOFSDBINDIR)/virtiofsd-x86_64
 # Default DAX mapping cache size in MiB
-DEFVIRTIOFSCACHESIZE := 8192
+DEFVIRTIOFSCACHESIZE := 1024
```
Comment (grahamwhaley) [inline Makefile]: Some hint of why the size change would be nice ;-)

File: cli/config/configuration-nemu.toml.in [hunk 3]
```diff
@@ -10,7 +10,7 @@
 # XXX:   Name: @PROJECT_NAME@
 # XXX:   Type: @PROJECT_TYPE@
 
-[hypervisor.nemu]
+[hypervisor.qemu]
```
Comment (grahamwhaley) [inline cli/config/configuration-nemu.toml.in]: I presume this was just fixing up a typo, and not matching change is needed in the runtime parser or binary deployment names for instance?

**Comment by grahamwhaley** [comment]
There are some CI failures. I suspect they will be unrelated, but need checking. For instance, from 18.04:

```
• Failure [32.884 seconds]
[Serial Test] package manager update test
/tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:32
  check apt-get update and upgrade
  /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:54
    should not fail [It]
    /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:55

    Expected
        <int>: 100
    to be zero-valued

    /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:61
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:

[Fail] [Serial Test] package manager update test check apt-get update and upgrade [It] should not fail 
/tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/package_manager_test.go:61

Ran 9 of 249 Specs in 98.779 seconds
FAIL! -- 8 Passed | 1 Failed | 0 Pending | 240 Skipped --- FAIL: TestIntegration (98.78s)
FAIL
```

File: cli/config/configuration-nemu.toml.in (hunk 3 above)
Comment (jodh-intel) [reply cli/config/configuration-nemu.toml.in]: It looks like this is in fact by design as the commit mentions:

> use the qemu hypervisor type for configuration template

@egernst - could you add a comment in the file explaining this though please?

File: Makefile (hunk 2 above)
Comment (egernst) [reply Makefile]: done (in commit message)

**Comment by grahamwhaley** [comment]
/test
18.04 CI still failed - but, I think that might be the missing kernel patch issue we had - which, I think we now fixed? Let me spin this to check...

**Pull request merged**